utils/src/time.rs

```diff
@@ -3,6 +3,112 @@
+/// - Values that would cause integer overflow return an error
+///
+/// # Arguments
+///
+/// * `s` - A string slice containing the duration with optional suffix
+///
+/// # Returns
+///
+/// * `Ok(Duration)` - Successfully parsed duration
+/// * `Err(String)` - Error message describing what went wrong (invalid format, overflow, etc.)
+///
+/// # Examples
+///
+/// ```
+/// # use commonware_utils::parse_duration;
+/// # use std::time::Duration;
+///
+/// // Different time units
+/// assert_eq!(parse_duration("500ms").unwrap(), Duration::from_millis(500));
+/// assert_eq!(parse_duration("30s").unwrap(), Duration::from_secs(30));
+/// assert_eq!(parse_duration("5m").unwrap(), Duration::from_secs(300));
+/// assert_eq!(parse_duration("2h").unwrap(), Duration::from_secs(7200));
+///
+/// // No suffix defaults to seconds
+/// assert_eq!(parse_duration("60").unwrap(), Duration::from_secs(60));
+///
+/// // Error cases
+/// assert!(parse_duration("invalid").is_err());
+/// assert!(parse_duration("10x").is_err());
+/// assert!(parse_duration("5minutes").is_err()); // Long forms not supported
+///
+/// // Overflow protection
+/// let max_hours = u64::MAX / 3600;
+/// assert!(parse_duration(&format!("{}h", max_hours)).is_ok());     // At limit
+/// assert!(parse_duration(&format!("{}h", max_hours + 1)).is_err()); // Overflow
+/// ```
+pub fn parse_duration(s: &str) -> Result<Duration, String> {
+    let s = s.trim();
+
+    // Handle milliseconds
+    if let Some(num_str) = s.strip_suffix("ms") {
+        let millis: u64 = num_str
+            .trim()
+            .parse()
+            .map_err(|_| format!("Invalid milliseconds value: '{num_str}'"))?;
+        return Ok(Duration::from_millis(millis));
+    }
+
+    // Handle hours
+    if let Some(num_str) = s.strip_suffix("h") {
+        let hours: u64 = num_str
+            .trim()
+            .parse()
+            .map_err(|_| format!("Invalid hours value: '{num_str}'"))?;
+        let seconds = hours
+            .checked_mul(3600)
+            .ok_or_else(|| format!("Hours value too large (would overflow): '{hours}'"))?;
+        return Ok(Duration::from_secs(seconds));
+    }
+
+    // Handle minutes
+    if let Some(num_str) = s.strip_suffix("m") {
+        let minutes: u64 = num_str
+            .trim()
+            .parse()
+            .map_err(|_| format!("Invalid minutes value: '{num_str}'"))?;
+        let seconds = minutes
+            .checked_mul(60)
+            .ok_or_else(|| format!("Minutes value too large (would overflow): '{minutes}'"))?;
+        return Ok(Duration::from_secs(seconds));
+    }
+
+    // Handle seconds
+    if let Some(num_str) = s.strip_suffix("s") {
+        let secs: u64 = num_str
+            .trim()
+            .parse()
+            .map_err(|_| format!("Invalid seconds value: '{num_str}'"))?;
+        return Ok(Duration::from_secs(secs));
+    }
+
+    // No suffix - parse as seconds
```

Comment:
nit: I wonder if its worth erroring on the base case? Seems reasonable to require some specificity? idk.